Source/Evolve/WarpXEvolve.cpp

```diff
@@ -176,6 +176,8 @@ WarpX::Evolve (int numsteps)
         if (evolve_scheme == EvolveScheme::Explicit) {
```

Comment:
I would suggest a more correct way of fixing this is to move this whole block of code to the end of the time step after the electrostatic field calculation (and before the diagnostics). This is suggested in the ancient PR #1751. This fixes two things. At that point the particle boundary conditions would have been handled anyway, and also the fields updated to the correct time level for the velocity push.

Reply:
Unfortunately, this causes many CI tests to fail. I can work through the tests and update the benchmarks, making sure that nothing is actually broken.